runtime/src/main/java/org/corfudb/runtime/object/transactions/Transaction.java

```diff
@@ -54,22 +54,32 @@ public boolean isLoggingEnabled() {
     }
 
     /**
-     * Verifies that this transaction  has a valid snapshot and context.
+     * Verifies that this transaction has a valid snapshot and context.
      */
     private void verify() {
         final AbstractTransactionalContext parentCtx = TransactionalContext.getCurrentContext();
         if (parentCtx != null) {
+            // In a nested transaction, the runtime should be the same as the parent's
             if (parentCtx.getTransaction().getRuntime() != this.runtime) {
-                throw new IllegalStateException("Attempting to nest transactions from different clients!");
+                throw new IllegalStateException(
+                        "Attempting to nest transactions from different clients!");
             }
-            // If we're in a nested transaction, the first read timestamp
-            // needs to come from the root.
+
+            // In a nested transaction, the transaction type should be the same as the parent's
+            if (type != parentCtx.getTransaction().getType()) {
+                String msg = String.format("Attempting to nest transactions with a " +
+                                "different transaction type, parent type=%s, child type=%s",
```

Comment:
Maybe nested instead of child transaction?

Reply:
Done.